dropshot/src/server.rs

```diff
@@ -1061,6 +1082,74 @@ impl BuildError {
     }
 }
 
+/// Start configuring a Dropshot server
+#[derive(Debug)]
+pub struct ServerBuilder<C: ServerContext> {
+    // required caller-provided values
+    private: C,
+    log: Logger,
+    api: DebugIgnore<ApiDescription<C>>,
+
+    // optional caller-provided values
+    config: ConfigDropshot,
+    tls: Option<ConfigTls>,
+}
+
+impl<C: ServerContext> ServerBuilder<C> {
+    /// Start configuring a new Dropshot server
+    ///
+    /// * `api`: the API to be hosted on this server
+    /// * `private`: your private data that will be made available in
+    ///   `RequestContext`
+    /// * `log`: a slog logger for all server events
+    pub fn new(
+        api: ApiDescription<C>,
```

Comment:
what do you think about having ApiDescriptions added as an `fn api(mut self, ...` anticipating having multiple APIs? i.e. the much discussed "experimental" API?

Reply:
I had that previously and I'd still be game to do it.  The reason I undid it is that I find it slightly worse now (because it doesn't communicate "you need exactly one of these") and that I usually find when I do something like this "to get ready" for some future change, it's never quite right by the time I go do the later thing anyway so it just made things worse in the meantime and made more work later.

Reply:
Fair enough; we can do that when we do it.